pkgs/by-name/so/sourcegit/package.nix

```diff
@@ -0,0 +1,109 @@
+{
+  lib,
+  stdenv,
+  buildDotnetModule,
+  fetchFromGitHub,
+  dotnetCorePackages,
+  copyDesktopItems,
+  makeDesktopItem,
+  libicns,
+
+  libXcursor,
+  libXext,
+  libXi,
+  libXrandr,
+
+  git,
+  xdg-utils,
+}:
+
+buildDotnetModule rec {
+  pname = "sourcegit";
+  version = "8.44";
+
+  src = fetchFromGitHub {
+    owner = "sourcegit-scm";
+    repo = "sourcegit";
+    tag = "v${version}";
+    hash = "sha256-A8Pn2JYT7WD5iUvwDCht9d2CiWdAdjX499Ic0EoyExw=";
+  };
+
+  patches = [ ./fix-darwin-git-path.patch ];
+
+  dotnet-sdk = dotnetCorePackages.sdk_9_0;
+  dotnet-runtime = dotnetCorePackages.runtime_9_0;
+
+  nugetDeps = ./deps.json;
+
+  projectFile = [ "src/SourceGit.csproj" ];
+
+  executables = [ "SourceGit" ];
+
+  dotnetFlags = [
+    "-p:DisableUpdateDetection=true"
+    "-p:DisableAOT=true"
+  ];
+
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    copyDesktopItems
+    libicns
+  ];
+
+  # these are dlopen-ed at runtime
+  # libXi is needed for right-click support
+  # not sure about what the other ones are needed for, but I'll include them anyways
+  runtimeDeps = [
+    libXcursor
+    libXext
+    libXi
+    libXrandr
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "SourceGit";
+      exec = "SourceGit";
+      icon = "SourceGit";
+      desktopName = "SourceGit";
+      terminal = false;
+      comment = meta.description;
+    })
+  ];
+
+  # add fallback binaries to use if the user doesn't already have them in their PATH
+  preInstall = ''
+    makeWrapperArgs+=(
+      --suffix PATH : ${
+        lib.makeBinPath [
+          git
+          xdg-utils
+        ]
+      }
+    )
+  '';
+
+  postInstall =
+    lib.optionalString stdenv.hostPlatform.isLinux ''
+      icns2png -x build/resources/App.icns
+      for f in App_*x32.png; do
+          res=''${f//App_}
+          res=''${res//x32.png}
+          install -Dm644 $f "$out/share/icons/hicolor/$res/apps/SourceGit.png"
+      done
+    ''
+    + lib.optionalString stdenv.hostPlatform.isDarwin ''
+      install -Dm644 build/resources/app/App.icns $out/Applications/SourceGit.app/Contents/Resources/App.icns
+      sed "s/SOURCE_GIT_VERSION/${version}/g" build/resources/app/App.plist > $out/Applications/SourceGit.app/Contents/Info.plist
```

Comment:
This will make the version very annoying to override. Would you mind using the `finalAttrs` pattern here such that we can override `version` and `src` and have the rest of the expression use the right values automatically?

Reply:
AFAIK buildDotnetModule doesnt support finalAttrs

Reply:
Also, this is just some metadata and not the end of the world if it becomes outdated.

Reply:
But, now that think about it: we could use $version (bash variable substitution) instead which would essentially be the same as finalAttrs

Reply:
It does support it ;) 

https://github.com/NixOS/nixpkgs/pull/331398

Reply:
Couldn't resist looking at the implementation and leaving some feedback: https://github.com/NixOS/nixpkgs/pull/331398#discussion_r1903378702

I have also tried to add a "good" fixed-point system for `buildRustPackage`, and I'm still searching for an "ideal" solution.